File: src/main/Python/boj_step6/boj1316.py
import sys
cin = sys.stdin.readline

N = int(cin())
set = set()
cnt = 0
for _ in range(N):
    is_group = True
    set.clear()
    s = cin().strip()
    for i, c in enumerate(s):
        if c in set and s[i - 1] != c:
            is_group = False
            break
        set.add(c)
    if is_group: cnt += 1
print(cnt)

File: src/main/Cpp/boj_step6/boj1316.cpp
#include <iostream>
#include <unordered_set>
#include <string>
using namespace std;

int main() {
    int N;
    cin >> N;
    cin.ignore();
    int cnt = 0;
    for (int k = 0; k < N; ++k) {
        bool is_group = true;
        unordered_set<char> key_set;
        string s;
        cin >> s;
        for (int i = 0; i < s.length(); ++i) {
            if (key_set.find(s[i]) != key_set.end() && s[i - 1] != s[i]) {
                is_group = false;
                break;
            }
            key_set.insert(s[i]);
        }
        if (is_group) cnt++;
    }
    cout << cnt << endl;
}